### PR TITLE
Fix for issue #64

### PR DESCRIPTION
**Pull request by this**
to fix issue #64

### CODE REVIEW

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/StartupListener.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    private AppDeployer appDeployer;
+
+    @Reference(name = "httpConnector",
+               service = HttpConnector.class,
+               cardinality = ReferenceCardinality.AT_LEAST_ONE,
+               policy = ReferencePolicy.DYNAMIC,
+               unbind = "unsetHttpConnector")
+    public void setHttpConnector(HttpConnector httpConnector) {
+        log.debug("HttpConnector '" + httpConnector.getClass().getName() + "' registered.");
```
Comment (jsdjayanga) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/StartupListener.java]: can use parameterized logging

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/UUFServer.java [hunk 2]
```diff
@@ -18,14 +18,181 @@
...
+                   "componentName=wso2-uuf-server"
+           }
+)
+public class UUFServer implements Server, AppDeployer, RequiredCapabilityListener {
 
     private static final boolean DEV_MODE_ENABLED;
+    private static final Logger log = LoggerFactory.getLogger(UUFServer.class);
 
     static {
         DEV_MODE_ENABLED = Boolean.parseBoolean(System.getProperties().getProperty("devmode", "false"));
```
Comment (jsdjayanga) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/UUFServer.java]: Check whether we can use Boolean.getBoolean() here

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/io/ArtifactAppDeployer.java [hunk 3]
```diff
@@ -19,199 +19,97 @@
...
         }
 
-        eventPublisher.publish(httpConnector -> httpConnector
-                .registerConnection(new ServerConnection(appNameContextPath.getRight(), this)));
-
-        pendingToDeployArtifacts.put(appNameContextPath.getRight(),
-                                     new AppArtifact(appNameContextPath.getLeft(), artifact));
-        log.debug("UUF app '" + appNameContextPath.getLeft() + "' added to the pending deployments list.");
-        return appNameContextPath.getLeft();
+        list.filter(Files::isDirectory).forEach(appPath -> {
```
Comment (jsdjayanga) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/io/ArtifactAppDeployer.java]: Code formatting

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/io/util/ZipArtifactHandler.java [hunk 4]
```diff
@@ -40,61 +38,59 @@
...
-    public static Path unzip(Artifact zipArtifact, String appName) {
+    public static Path unzip(String appName, Path zipFile) {
         Path appDirectory = TEMP_DIRECTORY.resolve(appName);
         if (Files.exists(appDirectory)) {
-            // A directory already exists in the tmp folder with the same app name, delete it before unzipping
-            // the new app.
+            /* A directory already exists in the tmp folder with the same app name, delete it before unzipping the
+            new app. */
             try {
                 FileUtils.deleteDirectory(appDirectory.toFile());
```
Comment (jsdjayanga) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/io/util/ZipArtifactHandler.java]: Can you check whether this operation works fine in Windows

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/StartupListener.java (hunk 1 above)
Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/StartupListener.java]: fixed in e08e8667d2da06165fe958da2fbe927c506256da

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/UUFServer.java (hunk 2 above)
Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/UUFServer.java]: 63bcd549a62d334da4a0ca909fb43aa97defc34b

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/io/util/ZipArtifactHandler.java (hunk 4 above)
Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/io/util/ZipArtifactHandler.java]: works in Windows

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/io/ArtifactAppDeployer.java (hunk 3 above)
Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/io/ArtifactAppDeployer.java]: fixed in abefbc446a2a1d0796cbe8eaa37972cc38a0e468

**Pull request merged**